Hackney.Shared.Finance/Infrastructure/Specifications/Specification.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq.Expressions;
+
+namespace Hackney.Shared.Finance.Infrastructure.Specifications
+{
+    public abstract class Specification<T> : ISpecification<T> where T : class
+    {
+        protected Specification([NotNull] Expression<Func<T, bool>> criteria)
+        {
+            Criteria = criteria;
+        }
+
+        public Expression<Func<T, bool>> Criteria { get; }
+    }
+}
```

Comment:
I'm not familiar with this object. Is this something that comes with Nuget template, or is this used anywhere?